block/manager.go

```diff
@@ -293,14 +315,21 @@ func (m *Manager) syncFromSettlement() error {
 		// TODO: separate between fresh rollapp and non-registered rollapp
 		return err
 	}
+
 	m.LastSubmittedHeight.Store(res.EndHeight)
-	err = m.syncToTargetHeight(res.EndHeight)
 	m.UpdateTargetHeight(res.EndHeight)
```

Comment:
this is a general note on the `UpdateTargetHeight`. we're calling this from multiple places (like on new p2p block etc) but not necessarily with the latest height.
i.e this should be equal to the latest height we know (from both settlement and p2p) but from p2p for example we update this height before we check if the block we recived is at heigher height than we have. 
WE should revise all callsites to this and see we only call it when we know the new height is >= store.NextHeight()

Reply:
done